public/app/features/plugins/datasource_srv.ts

```diff
@@ -324,6 +325,11 @@ export class DatasourceSrv implements DataSourceService {
       };
     });
   }
+
+  async reload() {
+    const settings = await getBackendSrv().get('/api/frontend/settings');
+    getDatasourceSrv().init(settings.datasources, settings.defaultDatasource);
```

Comment:
**Question:** couldn't we just use `this.init(settings.datasources, settings.defaultDatasource)`?

As far as I understand this is the singleton instance that get's returned by `getDatasourceSrv()`, but can be that I am wrong.

Reply:
@leventebalogh true, would be the same. Changed: https://github.com/grafana/grafana/pull/44217/commits/22ead3fee1dd032eff7808f71ad89887afb6bb5f